ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/QueryStreamWriter.java

```diff
@@ -124,14 +126,22 @@ public synchronized void close() {
   }
 
   private StreamedRow buildHeader() {
+    final QueryId queryId = queryMetadata.getQueryId();
+
     // Push queries only return value columns, but query metadata schema includes key and meta:
     final LogicalSchema storedSchema = queryMetadata.getLogicalSchema();
 
-    final Builder actualSchemaBuilder = LogicalSchema.builder();
+    final Builder projectionSchema = LogicalSchema.builder();
+
+    storedSchema.value().forEach(projectionSchema::valueColumn);
 
-    storedSchema.value().forEach(actualSchemaBuilder::valueColumn);
+    return StreamedRow.header(queryId, projectionSchema.build());
+  }
 
-    return StreamedRow.header(NO_QUERY_ID, actualSchemaBuilder.build());
+  private StreamedRow buildRow(final KeyValue<List<?>, GenericRow> row) {
```

Comment:
Now that we're no longer returning the key as part of the API, is there any point in continuing to pipe it in? I'm wondering if these changes can be removed, along with KeyValue and StructKeyUtil#asList.

Reply:
We still need the key.

Deletes will come through with `row.value` as `null`.   The code then needs to build a new `GenericRow` to send to the client using `row.key`.  For example, consider these two key-values, where the second represents a deletion of the first:

```sql
-- Given query, where ID is the primary key:
SELECT ID, COL0 FROM FOO EMIT CHANGES;
```

| key | value |
|----|-------|
| 10 | [10, 1] |
| 10 | null |

The code needs to translate the above two key-values posted to the blocking queue into a `GenericRow` that can be sent to the client.  When the `value` is not `null` we can just send the `value`. However, when `value` is `null` we must build a new `GenericRow` with any primary key columns set, which in this case would be `[10, null]`.

Once all three APIs are handling tombstones we should be able to move this logic to the producing side of the queue and just queue the `StreamedRow`, e.g. we'd just queue:

| value |
| {"columns": [10,1]"} |
| {"columns": [10,null]", "tombstone": true} |